components/server/src/workspace/workspace-starter.ts

```diff
@@ -674,8 +691,13 @@ export class WorkspaceStarter {
             }
         }
         if (projectEnvVars.length > 0) {
-            const projectEnvVarsWithValues = await this.projectDB.getProjectEnvironmentVariableValues(projectEnvVars);
-            allEnvVars = allEnvVars.concat(projectEnvVarsWithValues);
+            // Instead of using an access guard for Project environment variables, we let Project owners decide whether
+            // a variable should be:
+            //   - exposed in all workspaces (even for non-Project members when the repository is public), or
+            //   - censored from all workspaces (even for Project members)
+            const publicPrjEnvVars = projectEnvVars.filter(variable => !variable.censored);
```

Comment:
Since you do need a few special env vars in this file, I'm okay with moving the censoring/filtering code here (although the "hydration" with values can still happen right above).

One question though: Does it ever make sense to allow private registries to not be `censored`? (I.e. expose the value in workspaces.) If it never makes sense, I guess we could always filter out private registries here. 💭

Reply:
> Does it ever make sense to allow private registries to not be censored? (I.e. expose the value in workspaces.) 

It does in a future iteration when we'd merge this value with the `~/docker/config.json` file, hence provide access to private repos in the workspace.

> If it never makes sense, I guess we could always filter out private registries here. 💭

If we did this now, we'd introduce another failure mode - without telling anyone. I.e. if a user were to make the variable "visible" things just wouldn't work, and they wouldn't know why.